[ews-build.webkit.org] Make Contributors.load asynchronous

JonWBedard opened the pull request:
#### 25a06d784589d2da3d4ebfb93a82480b273bed55
<pre>
[ews-build.webkit.org] Make Contributors.load asynchronous
<a href="https://bugs.webkit.org/show_bug.cgi?id=254470">https://bugs.webkit.org/show_bug.cgi?id=254470</a>
rdar://107228856

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHub.email_for_owners): Make function asynchronous.
(Contributors.load_from_github): Ditto.
(Contributors.load): Ditto.
(ConfigureBuild.add_pr_details): Invoke Contributors.load as an
asynchronous function.
(ValidateChange.send_email_for_github_failure): Ditto.
(ValidateCommitterAndReviewer.run): Ditto.
(AnalyzeCompileWebKitResults.send_email_for_new_build_failure): Ditto.
(AnalyzeLayoutTestsResults.send_email_for_new_test_failures):Ditto.
(AddReviewerMixin.gitCommitEnvironment): Make asynchronous function
to invoke Contributors.load as aan asynchronousn function.
(AddReviewerToCommitMessage.run):Invoke gitCommitEnvironment as an
asynchronousn function.
(ValidateCommitMessage.run): Invoke Contributors.load as an
asynchronousn function.
(Canonicalize.run): Ditto.

Canonical link: <a href="https://commits.webkit.org/263086@main">https://commits.webkit.org/263086@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1967238c3f250006d843fa059c094f29570b549c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3570 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3616 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3757 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4995 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3844 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3549 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3710 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3658 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/3101 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3613 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3860 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3217 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4817 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1366 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3191 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/3093 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/3167 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3250 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4582 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3627 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2931 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/3533 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3167 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3189 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3195 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/410 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3446 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
r+ assuming this is tested.

JonWBedard committed:
[ews-build.webkit.org] Make Contributors.load asynchronous
https://bugs.webkit.org/show_bug.cgi?id=254470
rdar://107228856

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHub.email_for_owners): Make function asynchronous.
(Contributors.load_from_github): Ditto.
(Contributors.load): Ditto.
(ConfigureBuild.add_pr_details): Invoke Contributors.load as an
asynchronous function.
(ValidateChange.send_email_for_github_failure): Ditto.
(ValidateCommitterAndReviewer.run): Ditto.
(AnalyzeCompileWebKitResults.send_email_for_new_build_failure): Ditto.
(AnalyzeLayoutTestsResults.send_email_for_new_test_failures):Ditto.
(AddReviewerMixin.gitCommitEnvironment): Make asynchronous function
to invoke Contributors.load as aan asynchronousn function.
(AddReviewerToCommitMessage.run):Invoke gitCommitEnvironment as an
asynchronousn function.
(ValidateCommitMessage.run): Invoke Contributors.load as an
asynchronousn function.
(Canonicalize.run): Ditto.

Canonical link: https://commits.webkit.org/263086@main
